spec.html

```diff
@@ -39569,18 +39571,16 @@ <h1>Array.prototype.toLocaleString ( [ _reserved1_ [ , _reserved2_ ] ] )</h1>
         <p>The meanings of the optional parameters to this method are defined in the ECMA-402 specification; implementations that do not include ECMA-402 support must not use those parameter positions for anything else.</p>
         <p>This method performs the following steps when called:</p>
         <emu-alg>
-          1. Let _array_ be ? ToObject(*this* value).
-          1. Let _len_ be ? LengthOfArrayLike(_array_).
+          1. Let _O_ be ? ToObject(*this* value).
+          1. Let _len_ be ? LengthOfArrayLike(_O_).
           1. Let _separator_ be the implementation-defined list-separator String value appropriate for the host environment's current locale (such as *", "*).
           1. Let _R_ be the empty String.
           1. Let _k_ be 0.
           1. Repeat, while _k_ &lt; _len_,
-            1. If _k_ > 0, then
-              1. Set _R_ to the string-concatenation of _R_ and _separator_.
-            1. Let _nextElement_ be ? Get(_array_, ! ToString(𝔽(_k_))).
-            1. If _nextElement_ is neither *undefined* nor *null*, then
-              1. Let _S_ be ? ToString(? Invoke(_nextElement_, *"toLocaleString"*)).
-              1. Set _R_ to the string-concatenation of _R_ and _S_.
+            1. If _k_ > 0, set _R_ to the string-concatenation of _R_ and _separator_.
+            1. Let _element_ be ? Get(_O_, ! ToString(𝔽(_k_))).
+            1. If _element_ is either *undefined* or *null*, let _S_ be the empty String; otherwise, let _S_ be ? ToString(? Invoke(_element_, *"toLocaleString"*)).
+            1. Set _R_ to the string-concatenation of _R_ and _S_.
```

Comment:
I don't like these changes. Let's change `Array.prototype.join` to match this instead.

Reply:
Sure, done :+1: